From tsutomu.owa@toshiba.co.jp Mon May 14 15:29:17 2007
Date: Mon, 14 May 2007 15:29:17 +0900
From: Tsutomu OWA <tsutomu.owa@toshiba.co.jp>
To: linuxppc-dev@ozlabs.org, linux-kernel@vger.kernel.org
Cc: mingo@elte.hu, tglx@linutronix.de
Subject: Re: [patch 3/4] powerpc 2.6.21-rt1: add a need_resched_delayed() check


Add a need_resched_delayed() check.
This was pointed by Sergei Shtylyov; 
  http://ozlabs.org/pipermail/linuxppc-dev/2007-March/033148.html

Signed-off-by: Tsutomu Owa <tsutomu.owa@toshiba.co.jp>
-- owa

---
 arch/powerpc/kernel/idle.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

Index: linux-2.6.23.1-rt5/arch/powerpc/kernel/idle.c
===================================================================
--- linux-2.6.23.1-rt5.orig/arch/powerpc/kernel/idle.c
+++ linux-2.6.23.1-rt5/arch/powerpc/kernel/idle.c
@@ -75,7 +75,9 @@ void cpu_idle(void)
 				local_irq_disable();
 
 				/* check again after disabling irqs */
-				if (!need_resched() && !cpu_should_die())
+				if (!need_resched() &&
+				    !need_resched_delayed() &&
+				    !cpu_should_die())
 					ppc_md.power_save();
 
 				local_irq_enable();
